gobblin-metrics-libs/gobblin-metrics-base/src/main/avro/GaaSObservabilityEventExperimental.avsc

```diff
@@ -188,6 +188,38 @@
           }
         }
       ]
-    }]
+    },
+    {
+      "name": "datasetsWritten",
+      "type": [
+        "null",
+        {
+          "type": "array",
+          "items": {
+            "type": "record",
+            "name": "DatasetMetric",
+            "doc": "DatasetMetric contains bytes and records written by Gobblin writers for the dataset URN.",
+            "fields": [
+              {
+                "name": "datasetUrn",
+                "type": "string",
+                "doc": "URN of the dataset"
+              },
+              {
+                "name": "bytesWritten",
+                "type": "long",
+                "doc": "Number of bytes written for the dataset"
```

Comment:
which jobs is this applicable to?  e.g. could it work for retention-release?  what about for pulling record-by-record from a CRM system and writing a subset of the records fields to a relational DB?  

how to measure the number of bytes "written" to the DB... maybe approximate from the char count of the stringified SQL statement?

(the questions I'm unclear on are probably worth anticipating from those looking at the "doc" strings here)

Reply:
For the DB writers it's not supported, for `JdbcWriter they support records written but for bytesWritten it's implemented as below.
```
  /**
   * This is not supported for JDBC writer.
   * {@inheritDoc}
   * @see org.apache.gobblin.writer.DataWriter#bytesWritten()
   */
  @Override
  public long bytesWritten() throws IOException {
    return -1L;
  }
  ```

Reply:
I'll add it to the doc for bytesWritten that it can be set to -1 for unsupported

Reply:
For jobs that this is applicable, it's not usable for retention but it should be compatible with any writer that records its records/bytes written, e.g. Distcp, Hdfs -> Mysql. So it's dependent on the writer implementation, but I can see that almost every Gobblin writer implements `recordsWritten()` and `bytesWritten()` as its enforced in the `Writer` interface.